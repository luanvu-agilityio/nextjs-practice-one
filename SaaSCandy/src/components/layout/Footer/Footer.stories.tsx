import type { Meta, StoryObj } from '@storybook/nextjs';
import { Footer } from './index';

const meta: Meta<typeof Footer> = {
  title: 'Layout/Footer',
  component: Footer,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;
type Story = StoryObj<typeof Footer>;

export const Default: Story = {
  render: () => <Footer />,
};

export const Mobile: Story = {
  parameters: {
    viewport: {
      defaultViewport: 'mobile1',
    },
  },
  render: () => <Footer />,
};
