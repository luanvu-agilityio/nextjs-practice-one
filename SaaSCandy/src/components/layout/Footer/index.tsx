'use client';

import { Button, Heading, IconButton, Typography } from '@/components/common';
import LogoIcon from '@/components/icons/Logo';
import { NAV_LINKS, SOCIAL_LINKS } from '@/constants';

import Link from 'next/link';
import { JSX } from 'react';

export default function Footer(): JSX.Element {
  return (
    <footer className='bg-dark-blue w-full '>
      <div className='max-w-[1296px] mx-auto px-0 py-17.5 flex justify-between gap-33'>
        <div className='w-76.5 flex flex-col gap-13.5'>
          <Link href='/' className='flex items-center gap-2 no-underline'>
            <LogoIcon className='w-8 h-8' />

            <div className='hidden sm:block'>
              <Heading
                as='h1'
                size='md'
                content={
                  <span className='font-secondary text-white text-3xl font-light'>
                    SaaS<span className='font-medium'>Candy</span>
                  </span>
                }
              />
            </div>
          </Link>
          <div className='flex flex-col gap-6.5'>
            <Typography className='text-lg text-white opacity-50 font-medium'>
              Rakon is a simple, elegant, and secure way to build your bitcoin
              and crypto portfolio.
            </Typography>
            <Typography className='text-white text-lg font-medium'>
              1989 Don Jackson Lane
              <br />
              <span className='text-lg text-blue-foreground font-medium'>
                Call us:{' '}
                <span className='text-orange-background'>808-956-9599</span>
              </span>
            </Typography>
          </div>
        </div>
        <div className='flex justify-between  gap-6 text-lg '>
          <div className='w-48'>
            <Heading
              as='h4'
              content='Services'
              className='text-white mb-10 font-medium text-lg'
            />
            <ul className='space-y-6 text-lg font-medium text-white hover:text-white opacity-50'>
              <li>
                <Link href='/services/edtech' className='hover:text-white'>
                  EdTech Apps
                </Link>
              </li>
              <li>
                <Link href='/services/ecommerce' className='hover:text-white'>
                  eCommerce Apps
                </Link>
              </li>
              <li>
                <Link href='/services/crm' className='hover:text-white'>
                  CRM Apps
                </Link>
              </li>
              <li>
                <Link href='/services/health' className='hover:text-white'>
                  Health Apps
                </Link>
              </li>
              <li>
                <Link href='/services/analytics' className='hover:text-white'>
                  Web Analytics Apps
                </Link>
              </li>
              <li>
                <Link href='/services/banking' className='hover:text-white'>
                  Banking Apps
                </Link>
              </li>
            </ul>
          </div>

          <div className='w-48'>
            <Heading
              as='h4'
              content='Company'
              className='text-white text-lg mb-10 font-medium '
            />
            <ul className='space-y-6 text-lg font-medium text-white opacity-50 hover:text-white'>
              {NAV_LINKS.map(l => (
                <li key={l.href}>
                  <Link href={l.href} className='hover:text-white'>
                    {l.label}
                  </Link>
                </li>
              ))}
            </ul>
          </div>

          <div className='w-104'>
            <Heading
              as='h4'
              content='Subscribe'
              className='text-white mb-10 font-medium text-lg'
            />
            <Typography className='text-lg font-medium text-white opacity-50 mb-4'>
              Subscribe to get the latest news from us
            </Typography>
            <form className='flex gap-2.5'>
              <input
                aria-label='Email'
                type='email'
                placeholder='Enter email address'
                className='flex-1 rounded-lg bg-transparent border border-gray-700 px-3 py-2 text-sm placeholder-gray-400'
              />
              <Button variant='primary' size='small'>
                Register
              </Button>
            </form>
          </div>
        </div>
      </div>

      <div className='relative max-w-[1296px] mx-auto pb-17.5'>
        <div className='text-lg font-medium text-white flex items-center justify-between'>
          <span className='opacity-50'>
            Â©{new Date().getFullYear()} - All Rights Reserved by GetNextJs
          </span>

          <div className='flex gap-6' aria-label='Social media links'>
            {SOCIAL_LINKS.map(s => {
              const IconComponent = s.icon;
              return (
                <IconButton key={s.href} variant='ghost'>
                  <Link
                    href={s.href}
                    target='_blank'
                    rel='noreferrer'
                    className='inline-block'
                  >
                    <IconComponent />
                  </Link>
                </IconButton>
              );
            })}
          </div>
        </div>
      </div>

      <Link
        href='#'
        className='fixed right-10 bottom-10 bg-orange-background text-white text-md font-semibold px-4 py-2.5 rounded-xl'
      >
        Get This Template
      </Link>
    </footer>
  );
}
