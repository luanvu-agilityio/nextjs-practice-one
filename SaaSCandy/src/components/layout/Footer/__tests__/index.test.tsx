import { fireEvent, render, screen } from '@testing-library/react';
import Footer from '@/components/layout/Footer';

jest.mock('@/constants', () => ({
  NAV_LINKS: [
    { href: '/about', label: 'About' },
    { href: '/contact', label: 'Contact' },
  ],
  SOCIAL_LINKS: [
    { href: 'https://facebook.com', icon: () => <div>Facebook</div> },
    { href: 'https://twitter.com', icon: () => <div>Twitter</div> },
  ],
}));

describe('Footer - Snapshot Tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it('should render brand section with logo and contact info', () => {
    render(<Footer />);

    expect(screen.getByText(/SaaS/)).toBeInTheDocument();
    expect(screen.getByText(/Candy/)).toBeInTheDocument();
    expect(screen.getByText('1989 Don Jackson Lane')).toBeInTheDocument();
    expect(screen.getByText('808-956-9599')).toBeInTheDocument();
  });

  it('should render copyright with current year', () => {
    render(<Footer />);

    const currentYear = new Date().getFullYear();
    expect(
      screen.getByText(new RegExp(currentYear.toString()))
    ).toBeInTheDocument();
  });

  it('should render fixed CTA button', () => {
    render(<Footer />);

    expect(screen.getByText(/Get Template/)).toBeInTheDocument();
  });
});

jest.mock('@/constants', () => ({
  NAV_LINKS: [
    { href: '/about', label: 'About' },
    { href: '/contact', label: 'Contact' },
  ],
  SOCIAL_LINKS: [
    { href: 'https://facebook.com', icon: () => <div>Facebook</div> },
    { href: 'https://twitter.com', icon: () => <div>Twitter</div> },
  ],
}));

describe('Footer - Interactive Tests', () => {
  it('should toggle services section on mobile', () => {
    render(<Footer />);

    const servicesButton = screen.getAllByText('Services')[0];

    // Click to open
    fireEvent.click(servicesButton);
    expect(screen.getByText('EdTech Apps')).toBeInTheDocument();
  });

  it('should toggle company section on mobile', () => {
    render(<Footer />);

    const companyButtons = screen.getAllByText('Company');
    const mobileButton = companyButtons[0];

    // Click to open
    fireEvent.click(mobileButton);
    expect(screen.getByText('About')).toBeInTheDocument();
    expect(screen.getByText('Contact')).toBeInTheDocument();
  });

  it('should toggle subscribe section on mobile', () => {
    render(<Footer />);

    const subscribeButton = screen.getAllByText('Subscribe')[0];

    // Click to open
    fireEvent.click(subscribeButton);
    expect(
      screen.getByPlaceholderText('Enter email address')
    ).toBeInTheDocument();
    expect(screen.getByText('Register')).toBeInTheDocument();
  });

  it('should render service links', () => {
    render(<Footer />);

    expect(screen.getAllByText('EdTech Apps').length).toBeGreaterThan(0);
    expect(screen.getAllByText('eCommerce Apps').length).toBeGreaterThan(0);
    expect(screen.getAllByText('CRM Apps').length).toBeGreaterThan(0);
  });

  it('should render social links', () => {
    render(<Footer />);

    expect(screen.getByText('Facebook')).toBeInTheDocument();
    expect(screen.getByText('Twitter')).toBeInTheDocument();
  });
});
