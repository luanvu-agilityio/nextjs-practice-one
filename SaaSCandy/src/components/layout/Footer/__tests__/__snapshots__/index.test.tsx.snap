// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Footer - Snapshot Tests should match snapshot 1`] = `
<div>
  <footer
    class="bg-dark-blue w-full"
  >
    <div
      class="max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 py-12 sm:py-16 lg:py-17.5"
    >
      <div
        class="flex flex-col lg:flex-row justify-between gap-12 lg:gap-33"
      >
        <div
          class="w-full lg:w-76.5 flex flex-col gap-8 lg:gap-13.5"
        >
          <a
            class="flex items-center gap-2 no-underline"
            href="/"
          >
            <svg
              class="w-8 h-8"
              fill="none"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="15.9999"
                cy="15.9999"
                fill="#FFD245"
                r="15.9999"
              />
              <mask
                height="32"
                id="mask0_2053_384"
                maskUnits="userSpaceOnUse"
                style="mask-type: alpha;"
                width="32"
                x="0"
                y="0"
              >
                <circle
                  cx="15.9999"
                  cy="15.9999"
                  fill="#FFD245"
                  r="15.9999"
                />
              </mask>
              <g
                mask="url(#mask0_2053_384)"
              >
                <circle
                  cx="19.3045"
                  cy="9.53418"
                  fill="#FFE486"
                  r="15.9999"
                />
                <circle
                  cx="28.3808"
                  cy="9.53418"
                  fill="#EE7B11"
                  r="15.9999"
                />
              </g>
              <circle
                cx="24.7011"
                cy="24.7013"
                fill="#1642B9"
                r="6.29869"
                stroke="white"
                stroke-width="2"
              />
            </svg>
            <div>
              <h1
                aria-level="1"
                class="text-2xl font-semibold"
                role="heading"
              >
                <span
                  class="font-secondary text-white text-2xl sm:text-3xl font-light"
                >
                  SaaS
                  <span
                    class="font-medium"
                  >
                    Candy
                  </span>
                </span>
              </h1>
            </div>
          </a>
          <div
            class="flex flex-col gap-6 lg:gap-6.5"
          >
            <p
              class="text-md sm:text-lg text-white opacity-50 font-medium"
              role="paragraph"
            >
              Rakon is a simple, elegant, and secure way to build your bitcoin and crypto portfolio.
            </p>
            <p
              class="text-md text-white sm:text-lg font-medium"
              role="paragraph"
            >
              1989 Don Jackson Lane
              <br />
              <span
                class="  sm:text-lg text-blue-foreground font-medium"
              >
                Call us:
                 
                <span
                  class="text-orange-background"
                >
                  808-956-9599
                </span>
              </span>
            </p>
          </div>
        </div>
        <div
          class="hidden lg:grid grid-cols-3 gap-6   sm:text-lg flex-1"
        >
          <div>
            <h4
              aria-level="4"
              class="text-white mb-10 font-medium sm:text-lg"
              role="heading"
            >
              Services
            </h4>
            <ul
              class="space-y-6   sm:text-lg font-medium text-white opacity-50"
            >
              <li>
                <a
                  class="hover:text-white hover:opacity-100 transition-opacity"
                  href="/services/edtech"
                >
                  EdTech Apps
                </a>
              </li>
              <li>
                <a
                  class="hover:text-white hover:opacity-100 transition-opacity"
                  href="/services/ecommerce"
                >
                  eCommerce Apps
                </a>
              </li>
              <li>
                <a
                  class="hover:text-white hover:opacity-100 transition-opacity"
                  href="/services/crm"
                >
                  CRM Apps
                </a>
              </li>
              <li>
                <a
                  class="hover:text-white hover:opacity-100 transition-opacity"
                  href="/services/health"
                >
                  Health Apps
                </a>
              </li>
              <li>
                <a
                  class="hover:text-white hover:opacity-100 transition-opacity"
                  href="/services/analytics"
                >
                  Web Analytics Apps
                </a>
              </li>
              <li>
                <a
                  class="hover:text-white hover:opacity-100 transition-opacity"
                  href="/services/banking"
                >
                  Banking Apps
                </a>
              </li>
            </ul>
          </div>
          <div>
            <h4
              aria-level="4"
              class="text-white sm:text-lg mb-10 font-medium"
              role="heading"
            >
              Company
            </h4>
            <ul
              class="space-y-6   sm:text-lg font-medium text-white opacity-50"
            >
              <li>
                <a
                  class="hover:text-white hover:opacity-100 transition-opacity"
                  href="/about"
                >
                  About
                </a>
              </li>
              <li>
                <a
                  class="hover:text-white hover:opacity-100 transition-opacity"
                  href="/contact"
                >
                  Contact
                </a>
              </li>
            </ul>
          </div>
          <div>
            <h4
              aria-level="4"
              class="text-white mb-10 font-medium sm:text-lg"
              role="heading"
            >
              Subscribe
            </h4>
            <p
              class="text-md sm:text-lg font-medium text-white opacity-50 mb-4"
              role="paragraph"
            >
              Subscribe to get the latest news from us
            </p>
            <form
              class="flex flex-col gap-2.5"
            >
              <input
                aria-label="Email"
                class="flex-1 rounded-lg bg-transparent border border-gray-700 px-3 py-2 text-sm placeholder-gray-400 text-white"
                placeholder="Enter email address"
                type="email"
              />
              <button
                class="inline-flex items-center justify-center font-semibold text-md leading-none tracking-normal rounded-xl transition-colors duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer bg-orange-background text-white hover:opacity-90 focus-visible:ring-primary px-4 py-2.5 w-full"
              >
                Register
              </button>
            </form>
          </div>
        </div>
        <div
          class="lg:hidden flex flex-col gap-4 w-full"
        >
          <div
            class="border-b border-gray-700"
          >
            <button
              class="w-full flex items-center justify-between py-4 text-white"
            >
              <h4
                aria-level="4"
                class="text-white font-medium"
                role="heading"
              >
                Services
              </h4>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down w-5 h-5 transition-transform"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
          <div
            class="border-b border-gray-700"
          >
            <button
              class="w-full flex items-center justify-between py-4 text-white"
            >
              <h4
                aria-level="4"
                class="text-white font-medium"
                role="heading"
              >
                Company
              </h4>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down w-5 h-5 transition-transform"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
          <div
            class="border-b border-gray-700"
          >
            <button
              class="w-full flex items-center justify-between py-4 text-white"
            >
              <h4
                aria-level="4"
                class="text-white font-medium"
                role="heading"
              >
                Subscribe
              </h4>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down w-5 h-5 transition-transform"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="relative mt-12 lg:mt-0 lg:pt-17.5"
      >
        <div
          class="  sm:text-lg font-medium text-white flex flex-col sm:flex-row items-center justify-between gap-6"
        >
          <span
            class="opacity-50 text-center sm:text-left"
          >
            ©
            2025
             - All Rights Reserved by GetNextJs
          </span>
          <div
            aria-label="Social media links"
            class="flex gap-4 sm:gap-6"
          >
            <button
              class="cursor-pointer inline-flex items-center justify-center rounded-full transition-colors focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-ring focus-visible:ring-offset-0 disabled:opacity-50 disabled:cursor-not-allowed button-animation bg-transparent border-none text-primary hover:opacity-90 w-6 h-6 p-4"
              data-testid="icon-button"
            >
              <a
                class="inline-block"
                href="https://facebook.com"
                rel="noreferrer"
                target="_blank"
              >
                <div>
                  Facebook
                </div>
              </a>
            </button>
            <button
              class="cursor-pointer inline-flex items-center justify-center rounded-full transition-colors focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-ring focus-visible:ring-offset-0 disabled:opacity-50 disabled:cursor-not-allowed button-animation bg-transparent border-none text-primary hover:opacity-90 w-6 h-6 p-4"
              data-testid="icon-button"
            >
              <a
                class="inline-block"
                href="https://twitter.com"
                rel="noreferrer"
                target="_blank"
              >
                <div>
                  Twitter
                </div>
              </a>
            </button>
          </div>
        </div>
      </div>
    </div>
    <a
      class="fixed right-4 bottom-4 sm:right-6 sm:bottom-6 lg:right-10 lg:bottom-10 bg-orange-background text-white text-sm sm:text-md font-semibold px-3 py-2 sm:px-4 sm:py-2.5 rounded-xl shadow-lg hover:bg-orange-600 transition-colors z-50"
      href="#"
    >
      <span
        class="hidden sm:inline"
      >
        Get This Template
      </span>
      <span
        class="sm:hidden"
      >
        Get Template
      </span>
    </a>
  </footer>
</div>
`;

exports[`Footer - Snapshot matches snapshot 1`] = `
<div>
  <footer
    class="bg-dark-blue w-full"
  >
    <div
      class="max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 py-12 sm:py-16 lg:py-17.5"
    >
      <div
        class="flex flex-col lg:flex-row justify-between gap-12 lg:gap-33"
      >
        <div
          class="w-full lg:w-76.5 flex flex-col gap-8 lg:gap-13.5"
        >
          <a
            href="/"
          >
            <svg
              class="w-8 h-8"
              data-testid="logo-icon"
            />
            <div>
              <h1>
                <span
                  class="font-secondary text-white text-2xl sm:text-3xl font-light"
                >
                  SaaS
                  <span
                    class="font-medium"
                  >
                    Candy
                  </span>
                </span>
              </h1>
            </div>
          </a>
          <div
            class="flex flex-col gap-6 lg:gap-6.5"
          >
            <p
              class="  sm:text-lg text-white opacity-50 font-medium"
            >
              Rakon is a simple, elegant, and secure way to build your bitcoin and crypto portfolio.
            </p>
            <p
              class="text-white   sm:text-lg font-medium"
            >
              1989 Don Jackson Lane
              <br />
              <span
                class="  sm:text-lg text-blue-foreground font-medium"
              >
                Call us:
                 
                <span
                  class="text-orange-background"
                >
                  808-956-9599
                </span>
              </span>
            </p>
          </div>
        </div>
        <div
          class="hidden lg:grid grid-cols-3 gap-6   sm:text-lg flex-1"
        >
          <div>
            <h4
              class="text-white mb-10 font-medium   sm:text-lg"
            >
              Services
            </h4>
            <ul
              class="space-y-6   sm:text-lg font-medium text-white opacity-50"
            >
              <li>
                <a
                  href="/services/edtech"
                >
                  EdTech Apps
                </a>
              </li>
              <li>
                <a
                  href="/services/ecommerce"
                >
                  eCommerce Apps
                </a>
              </li>
              <li>
                <a
                  href="/services/crm"
                >
                  CRM Apps
                </a>
              </li>
              <li>
                <a
                  href="/services/health"
                >
                  Health Apps
                </a>
              </li>
              <li>
                <a
                  href="/services/analytics"
                >
                  Web Analytics Apps
                </a>
              </li>
              <li>
                <a
                  href="/services/banking"
                >
                  Banking Apps
                </a>
              </li>
            </ul>
          </div>
          <div>
            <h4
              class="text-white   sm:text-lg mb-10 font-medium"
            >
              Company
            </h4>
            <ul
              class="space-y-6   sm:text-lg font-medium text-white opacity-50"
            >
              <li>
                <a
                  href="/about"
                >
                  About
                </a>
              </li>
              <li>
                <a
                  href="/contact"
                >
                  Contact
                </a>
              </li>
            </ul>
          </div>
          <div>
            <h4
              class="text-white mb-10 font-medium   sm:text-lg"
            >
              Subscribe
            </h4>
            <p
              class="  sm:text-lg font-medium text-white opacity-50 mb-4"
            >
              Subscribe to get the latest news from us
            </p>
            <form
              class="flex flex-col gap-2.5"
            >
              <input
                aria-label="Email"
                class="flex-1 rounded-lg bg-transparent border border-gray-700 px-3 py-2 text-sm placeholder-gray-400 text-white"
                placeholder="Enter email address"
                type="email"
              />
              <button
                class="w-full"
                variant="primary"
              >
                Register
              </button>
            </form>
          </div>
        </div>
        <div
          class="lg:hidden flex flex-col gap-4 w-full"
        >
          <div
            class="border-b border-gray-700"
          >
            <button
              class="w-full flex items-center justify-between py-4 text-white"
            >
              <h4
                class="text-white font-medium  "
              >
                Services
              </h4>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down w-5 h-5 transition-transform"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
          <div
            class="border-b border-gray-700"
          >
            <button
              class="w-full flex items-center justify-between py-4 text-white"
            >
              <h4
                class="text-white font-medium  "
              >
                Company
              </h4>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down w-5 h-5 transition-transform"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
          <div
            class="border-b border-gray-700"
          >
            <button
              class="w-full flex items-center justify-between py-4 text-white"
            >
              <h4
                class="text-white font-medium  "
              >
                Subscribe
              </h4>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down w-5 h-5 transition-transform"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="relative mt-12 lg:mt-0 lg:pt-17.5"
      >
        <div
          class="  sm:text-lg font-medium text-white flex flex-col sm:flex-row items-center justify-between gap-6"
        >
          <span
            class="opacity-50 text-center sm:text-left"
          >
            ©
            2025
             - All Rights Reserved by GetNextJs
          </span>
          <div
            aria-label="Social media links"
            class="flex gap-4 sm:gap-6"
          >
            <button
              variant="ghost"
            >
              <a
                href="https://facebook.com"
              >
                <svg
                  data-testid="facebook-icon"
                />
              </a>
            </button>
            <button
              variant="ghost"
            >
              <a
                href="https://twitter.com"
              >
                <svg
                  data-testid="twitter-icon"
                />
              </a>
            </button>
          </div>
        </div>
      </div>
    </div>
    <a
      href="#"
    >
      <span
        class="hidden sm:inline"
      >
        Get This Template
      </span>
      <span
        class="sm:hidden"
      >
        Get Template
      </span>
    </a>
  </footer>
</div>
`;
